patches/server/1032-Fix-BlockRedstoneEvent-for-tripwire-hooks.patch

```diff
@@ -0,0 +1,72 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Thu, 14 Sep 2023 18:52:39 +0200
+Subject: [PATCH] Fix BlockRedstoneEvent for tripwire hooks
+
+Previously the event was called for any state update even non redstone related.
+Now only two events are always at least fired, more when the events are cancelled.
+
+diff --git a/src/main/java/net/minecraft/world/level/block/TripWireHookBlock.java b/src/main/java/net/minecraft/world/level/block/TripWireHookBlock.java
+index 004dce26ff073f1de52a84cd425c4f60fdab5e50..2d09d5160408a2f94fa24de1dd61048f73b94ae1 100644
+--- a/src/main/java/net/minecraft/world/level/block/TripWireHookBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/TripWireHookBlock.java
+@@ -160,26 +160,48 @@ public class TripWireHookBlock extends Block {
+         flag5 &= flag4;
+         BlockState iblockdata3 = (BlockState) ((BlockState) this.defaultBlockState().setValue(TripWireHookBlock.ATTACHED, flag4)).setValue(TripWireHookBlock.POWERED, flag5);
+ 
++        boolean cancelledEmitterHook = false, cancelledReceiverHook = false; // Paper
+         if (j > 0) {
+             blockposition1 = pos.relative(enumdirection, j);
++            // Paper start
++            if (flag3 != flag5) {
++                org.bukkit.block.Block block = world.getWorld().getBlockAt(blockposition1.getX(), blockposition1.getY(), blockposition1.getZ());
++                int newCurrent = flag5 ? 15 : 0;
++
++                BlockRedstoneEvent eventRedstone = new BlockRedstoneEvent(block, flag3 ? 15 : 0, newCurrent);
++                world.getCraftServer().getPluginManager().callEvent(eventRedstone);
++
++                if (eventRedstone.getNewCurrent() != newCurrent) {
++                    cancelledReceiverHook = true;
++                }
++            }
++            if (!cancelledReceiverHook) { // always trigger two events even when the first hook current change is cancelled
++            // Paper end
+             Direction enumdirection1 = enumdirection.getOpposite();
+ 
+             world.setBlock(blockposition1, (BlockState) iblockdata3.setValue(TripWireHookBlock.FACING, enumdirection1), 3);
+             this.notifyNeighbors(world, blockposition1, enumdirection1);
+             this.emitState(world, blockposition1, flag4, flag5, flag2, flag3);
++            } // Paper
+         }
+ 
+         // CraftBukkit start
++        if (flag3 != flag5) { // Paper
```

Comment:
Probably some obfhelpers for `flag3` and `flag5` would be nice to have here, and in the if statement added above as well.